apps/consent-ui/src/components/common/ProgressHeader/index.tsx

```diff
@@ -70,38 +76,45 @@ const ProgressHeader = ({
 		<div>
 			{/* Tablet and Desktop screens */}
 			<div className={styles.header}>
-				{CONSENT_CATEGORIES.map((step, stepIndex) => (
-					<Fragment key={step}>
-						<div className={styles.step}>
-							{loading || stepIndex > completeStepIndex + 1 ? (
-								<Incomplete className={styles.incomplete} />
-							) : stepIndex === currentStepIndex ? (
-								<InProgress className={styles['in-progress']} />
-							) : (
-								<Success className={styles.success} />
+				{CONSENT_CATEGORIES.map((step, stepIndex) => {
+					const isInProgress = stepIndex === currentStepIndex;
+					const isComplete = stepIndex <= completeStepIndex;
+					const isIncomplete = loading || (!isComplete && !isInProgress);
+					return (
+						<Fragment key={step}>
+							{stepIndex !== 0 && (
+								<hr
+									className={clsx(
+										styles.divider,
+										!loading && (isComplete || isInProgress) && styles['completed-step'],
+									)}
+								/>
 							)}
-							<span className={styles['desktop-step-name']}>{consentCategoryDict[step]}</span>
-							<span className={styles['tablet-step-name']}>
-								{replaceParams(tabletProgressHeader, { step: stepIndex + 1 })}
-							</span>
-						</div>
-						{stepIndex != CONSENT_CATEGORIES.length - 1 && (
-							<hr
-								className={clsx(
-									styles.divider,
-									!loading && stepIndex <= completeStepIndex && styles['completed-step'],
+							<div className={styles.step}>
+								{isIncomplete ? (
+									<Incomplete className={styles.incomplete} />
+								) : isInProgress ? (
+									<InProgress className={styles['in-progress']} />
+								) : (
+									<Success className={styles.success} />
 								)}
-							/>
-						)}
-					</Fragment>
-				))}
+								<span className={styles['desktop-step-name']}>
+									{stepIndex + 1}. {consentCategoryDict[step]}
+								</span>
+								<span className={styles['tablet-step-name']}>
+									{replaceParams(tabletProgressHeader, { step: stepIndex + 1 })}
+								</span>
+							</div>
+						</Fragment>
+					);
+				})}
 			</div>
 			{/* Mobile screens */}
 			<div className={styles['mobile-header']}>
 				<InProgress className={styles['in-progress']} />
 				<span className={styles['mobile-step-name']}>
 					{replaceParams(mobileProgressHeader, {
-						currentStep,
+						currentStep: currentStepIndex,
```

Comment:
I think we still need the ` + 1` like you had in a previous commit on this PR, since the `currentStepIndex` starts at 0 so the mobile progress is off by 1 currently.
```suggestion
						currentStep: currentStepIndex + 1,
```

Reply:
good catch